examples/plugin.cpp

```diff
@@ -0,0 +1,196 @@
+		kimera_pipeline.fillSingleImuQueue(VIO::ImuMeasurement(datum->dataset_time, imu_raw_vals));
+
+		// If there is not cam data this func call, break early
+		if (!datum->img0.has_value() && !datum->img1.has_value()) {
+			return;
+		}
+
+#ifdef CV_HAS_METRICS
+		cv::metrics::setAccount(new std::string{std::to_string(iteration_no)});
+		iteration_no++;
+		if (iteration_no % 20 == 0) {
+			cv::metrics::dump();
+		}
+#else
+#warning "No OpenCV metrics available. Please recompile OpenCV from git clone --branch 3.4.6-instrumented https://github.com/ILLIXR/opencv/. (see install_deps.sh)"
+#endif
+
+		cv::Mat img0{*imu_cam_buffer->img0.value()};
+		cv::Mat img1{*imu_cam_buffer->img1.value()};
+
+		// VIOParams
+		VIO::CameraParams left_cam_info = kimera_pipeline_params.camera_params_.at(0);
+		VIO::CameraParams right_cam_info = kimera_pipeline_params.camera_params_.at(1);
+		kimera_pipeline.fillLeftFrameQueue(VIO::make_unique<VIO::Frame>(kimera_current_frame_id,
+																	datum->dataset_time,
+																	left_cam_info, img0));
+		kimera_pipeline.fillRightFrameQueue(VIO::make_unique<VIO::Frame>(kimera_current_frame_id,
+																	datum->dataset_time,
+																	right_cam_info, img1));
+
+		kimera_pipeline.spin();
+#ifndef NDEBUG
+    	std::cout << "SPIN FULL\n";
+#endif
+	}
+
+  	void pose_callback(const std::shared_ptr<VIO::BackendOutput>& vio_output) {
+
+		const auto& cached_state = vio_output->W_State_Blkf_;
+		const auto& w_pose_blkf_trans = cached_state.pose_.translation().transpose();
+		const auto& w_pose_blkf_rot = cached_state.pose_.rotation().quaternion();
+		const auto& w_vel_blkf = cached_state.velocity_.transpose();
+		const auto& imu_bias_gyro = cached_state.imu_bias_.gyroscope().transpose();
+		const auto& imu_bias_acc = cached_state.imu_bias_.accelerometer().transpose();
+		// Get the pose returned from SLAM
+		Eigen::Quaternionf quat = Eigen::Quaternionf{w_pose_blkf_rot(0), w_pose_blkf_rot(1), w_pose_blkf_rot(2), w_pose_blkf_rot(3)};
+		Eigen::Quaterniond doub_quat = Eigen::Quaterniond{w_pose_blkf_rot(0), w_pose_blkf_rot(1), w_pose_blkf_rot(2), w_pose_blkf_rot(3)};
+		Eigen::Vector3f pos  = w_pose_blkf_trans.cast<float>();
+
+		assert(isfinite(quat.w()));
+		assert(isfinite(quat.x()));
+		assert(isfinite(quat.y()));
+		assert(isfinite(quat.z()));
+		assert(isfinite(pos[0]));
+		assert(isfinite(pos[1]));
+		assert(isfinite(pos[2]));
+
+		_m_pose->put(new pose_type{
+		.sensor_time = imu_cam_buffer->time,
+		.position = pos,
+		.orientation = quat,
+		});
+
+		_m_imu_integrator_input->put(new imu_integrator_input{
+			.last_cam_integration_time = (double(imu_cam_buffer->dataset_time) / NANO_SEC),
+			.t_offset = -0.05,
+
+			.params = {
+				.gyro_noise = 0.00016968,
+				.acc_noise = 0.002,
+				.gyro_walk = 1.9393e-05,
+				.acc_walk = 0.003,
+				.n_gravity = Eigen::Matrix<double,3,1>(0.0, 0.0, -9.81),
+				.imu_integration_sigma = 1.0,
+				.nominal_rate = 200.0,
```

Comment:
Is this the IMU rate? If so, we need a way to deal with ZED, given that ZED's IMU runs at 500 Hz.